### PR TITLE
SOLR-14659: Remove restlet as dependency for the ManagedResource API

### PR DESCRIPTION
**Pull request by thelabdude**
# Description

Here’s a quick pass at keeping the existing ManagedResource API endpoints in place but without using Restlet. Now that I’ve gone through this code again, it seems like Restlet was more aspirational than technically required to support the current functionality. As I said on the mailing list, I don’t know the history of why Restlet was introduced so I can’t say much more about that. 

What I do know is that it’s not needed to support the ManagedResource endpoints we have today. In a nutshell, RestManager keeps a mapping from a PATH (e.g. /schema/analysis/stopwords/english) to a ManagedResource impl and Restlet is used to route REST requests for the PATH to a ManagedResource. There’s a ManagedEndpoint used as the shim between Restlet and the ManagedResource that’s just details ...

In this PR, I’ve simply replaced the Restlet routing logic with a SolrRequestHandler that resolves the ManagedResource for a PATH using the RestManager’s mapping. Put simply, plugging into the Restlet routing logic isn’t necessary for Managed Resources. Not to mention, there’s some rigid mapping to /schema in places, so I’m not certain we achieved the supposed goal of supporting dynamic mapping of path to resource that Restlet provides.

As has been discussed on the mailing list, we could just remove all of this ManagedResource code from master but that requires us to come up with some LTR specific request handler to allow managing models and features. However, the approach in this PR doesn’t feel too dirty to me and avoids breaking the existing APIs for stopwords / synonyms / LTR models & features.

I’m happy to go either way here but wanted to see what it would take to remove Restlet but keep the existing endpoints and this work accomplishes that.

# Solution

Please provide a short description of the approach taken to implement your solution.

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `master` branch.
- [x] I have run `./gradlew check`.
- [x] I have added tests for my changes.
- [ ] I have added documentation for the [Ref Guide](https://github.com/apache/lucene-solr/tree/master/solr/solr-ref-guide) (for Solr changes only).

### CODE REVIEW

**Comment by noblepaul** [comment]
I've given a patch to be applied over this branch

**Comment by dsmiley** [comment]
The title of this PR refers to SOLR-14766 which is about deprecation and/or removal of managed resources overall.
I think this PR should be retitled to link to https://issues.apache.org/jira/browse/SOLR-14659

**Comment by thelabdude** [comment]
Thanks @noblepaul ... that's much cleaner. Patch applied.

File: solr/core/src/java/org/apache/solr/rest/RestManager.java [hunk 1]
```diff
@@ -326,44 +327,46 @@ public void doInit() throws ResourceException {
           }
         }
       }    
-      
+
       if (managedResource == null) {
-        if (Method.PUT.equals(getMethod()) || Method.POST.equals(getMethod())) {
+        final String method = getSolrRequest().getHttpMethod();
+        if ("PUT".equals(method) || "POST".equals(method)) {
```
[the remark this answers is not in the document]
Comment (thelabdude) [reply solr/core/src/java/org/apache/solr/rest/RestManager.java:336]: String equals unsafe? Not sure I understand what this warning is trying to convey?

**Comment by thelabdude** [comment]
@noblepaul so you had SOLR-14659 with fix version 8.7. Given all the tests pass, I think we could apply these changes for 8.7 but I'd be more comfortable if we just left 8.x as-is and apply these changes to 9.x ... wdyt?

**Comment by noblepaul** [comment]
@thelabdude  if all the tests pass, I wish this get committed to 8.x itself.

**Comment by thelabdude** [comment]
Ok, I'm good with that, mainly just wanted some buy-in from others ;-) Unless there are any objections, I'll move ahead with merging to master and 8.x ... thanks for the help @noblepaul

File: solr/core/src/java/org/apache/solr/rest/RestManager.java [hunk 2]
```diff
@@ -16,15 +16,33 @@
  */
 package org.apache.solr.rest;
 
+import org.apache.solr.common.SolrException;
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/rest/RestManager.java:19]: I'd prefer you configure your IDE to keep java.* up front.  FWIW this is in the IntelliJ config in the project.

File: solr/core/src/java/org/apache/solr/rest/RestManager.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (dsmiley) [reply solr/core/src/java/org/apache/solr/rest/RestManager.java:336]: If you click the details link, it explains.  It's pretty wild what it suggests... it's a stretch IMO.  Good luck pulling that attack off.  Besides, this is the HTTP method (fixed vocab); it's not a param.

Our use of Muse here is very new.... we haven't tweaked `.muse/config.toml` yet but it needs some taming.  https://docs.muse.dev/docs/configuring-muse/

**Review by noblepaul — APPROVED** [review]
All good. Please go ahead and merge

File: solr/core/src/java/org/apache/solr/rest/RestManager.java (hunk 2 above)
Comment (thelabdude) [reply solr/core/src/java/org/apache/solr/rest/RestManager.java:19]: oops my bad, thanks for catching

**Comment by dweiss** [comment]
Thank you!

**Comment by noblepaul** [comment]
🍻
cheers

**Pull request merged**